packages/lib/src/vis/matrix/context.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useMemo, createContext } from 'react';
+import type { ReactNode } from 'react';
+import type { GridOnItemsRenderedProps } from 'react-window';
+
+import type { Size } from '../models';
+
+interface GridSettings {
+  rowCount: number;
+  columnCount: number;
+  cellSize: Size;
+  sticky: boolean;
+  cellFormatter: (row: number, col: number) => string;
+  setRenderedItems: (renderedItems: GridOnItemsRenderedProps) => void;
+}
+
+interface Props extends Omit<GridSettings, 'setRenderedItems'> {
+  children: ReactNode;
+}
+
+export const SettingsContext = createContext({} as GridSettings);
+export const RenderedItemsContext = createContext<
+  GridOnItemsRenderedProps | undefined
+>(undefined);
+
+function GridProvider(props: Props) {
+  const { rowCount, columnCount, cellSize, sticky, cellFormatter, children } =
+    props;
+
+  const [renderedItems, setRenderedItems] =
+    useState<GridOnItemsRenderedProps>();
```

Comment:
The `renderedItems` state, which I provide through its own, independent context as explained by Kent C. Dodds: https://kentcdodds.com/blog/how-to-optimize-your-context-value